### PR TITLE
Split `{Has=>{Get,Set}}Field`

### PR DESCRIPTION
**Pull request by akrmn**
Daml PR: https://github.com/digital-asset/daml/pull/18319
DA-GHC PR: https://github.com/digital-asset/ghc/pull/177

### CODE REVIEW

**Review by samuel-williams-da — APPROVED** [review]
I can't even find a single thing to note!
I assume we're okay breaking the backwards compatibility to before we made this change?
Do we have a test that ensures custom Get/SetField instances persist across data dependencies?

**Comment by akrmn** [comment]
> I assume we're okay breaking the backwards compatibility to before we made this change?

yes, it's the great 2.x->3.x backwards compatibility break :)

> Do we have a test that ensures custom Get/SetField instances persist across data dependencies?

there are some tests in `DA.Test.DataDependencies` for `HasField` which still work, but none for custom instances. I'll add some now, thank you :)

File: UNRELEASED.md [hunk 1]
```diff
@@ -29,3 +29,12 @@ and ensure that the release notes and documentation has been included as part of
...
 
 ## What’s New
+
+* The type class `HasField` (methods `getField` and `setField`) has been split
+  in two: `GetField` and `SetField`, each one with the correspondingly named method.
+
+* User-defined instances of the class `GetField` enable the use of dot-syntax
+  for field access, i.e. `rec.field`
+
+* User-defined instances of the class `SetField` enable the use of with-syntax
```
Comment (basvangijzel-DA) [inline UNRELEASED.md:39]: Could you add a small example on how to migrate an existing user defined method that uses HasField to GetFIeld/SetFIeld? Feel free to do this in a separate PR.

**Review by basvangijzel-DA — APPROVED** [review]
Looks good, thanks Moisés!

File: UNRELEASED.md (hunk 1 above)
Comment (akrmn) [reply UNRELEASED.md:63]: Added in a new commit

**Pull request merged**